specs/phase1/beacon-chain.md

```diff
@@ -166,252 +131,99 @@ class AttestationData(Container):
     # FFG vote
     source: Checkpoint
     target: Checkpoint
-    # Shard vote
-    shard: Shard
-    # Current-slot shard block root
-    shard_head_root: Root
-    # Shard transition root
-    shard_transition_root: Root
-```
-
-### Extended `Attestation`
-
-```python
-class Attestation(Container):
-    aggregation_bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]
-    data: AttestationData
-    signature: BLSSignature
-```
-
-### Extended `PendingAttestation`
-
-```python
-class PendingAttestation(Container):
-    aggregation_bits: Bitlist[MAX_VALIDATORS_PER_COMMITTEE]
-    data: AttestationData
-    inclusion_delay: Slot
-    proposer_index: ValidatorIndex
-    # Phase 1
-    crosslink_success: boolean
-```
-
-### Extended `IndexedAttestation`
-
-```python
-class IndexedAttestation(Container):
-    attesting_indices: List[ValidatorIndex, MAX_VALIDATORS_PER_COMMITTEE]
-    data: AttestationData
-    signature: BLSSignature
-```
-
-### Extended `AttesterSlashing`
-
-Note that the `attestation_1` and `attestation_2` have a new `IndexedAttestation` definition.
-
-```python
-class AttesterSlashing(Container):
-    attestation_1: IndexedAttestation
-    attestation_2: IndexedAttestation
+    # Shard header root
+    shard_header_root: Root
 ```
 
-### Extended `Validator`
+### `BeaconBlock`
 
 ```python
-class Validator(Container):
-    pubkey: BLSPubkey
-    withdrawal_credentials: Bytes32  # Commitment to pubkey for withdrawals
-    effective_balance: Gwei  # Balance at stake
-    slashed: boolean
-    # Status epochs
-    activation_eligibility_epoch: Epoch  # When criteria for activation were met
-    activation_epoch: Epoch
-    exit_epoch: Epoch
-    withdrawable_epoch: Epoch  # When validator can withdraw funds
-    # Custody game
-    # next_custody_secret_to_reveal is initialised to the custody period
-    # (of the particular validator) in which the validator is activated
-    # = get_custody_period_for_validator(...)
-    next_custody_secret_to_reveal: uint64
-    # TODO: The max_reveal_lateness doesn't really make sense anymore.
-    # So how do we incentivise early custody key reveals now?
-    all_custody_secrets_revealed_epoch: Epoch  # to be initialized to FAR_FUTURE_EPOCH
+class BeaconBlock(phase0.BeaconBlock):
+    shard_headers: List[Signed[ShardHeader], MAX_SHARD_HEADERS]
```

Comment:
A few comments here:

1. `shard_headers` is not set anywhere, thus `assert len(body.shard_headers) <= (...)` will always fail.
2. Since `BeaconBlockBody` contains a list of attestations, should this field even exist? Shouldn't `BeaconBlockBody.attestations` hold headers within attestations themselves?
3. Even if we keep `shard_headers`, shouldn't the size of `BeaconBlockBody.attestations` be increased to cater for shards? Currently `MAX_ATTESTATIONS =  128`